File: test/language/templates/java/templates/InstantiateViaSingleImportTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;

import java.io.IOException;

import org.junit.jupiter.api.Test;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_via_single_import.InstantiateViaSingleImport;
import templates.instantiate_via_single_import.pkg.Test_string;
import templates.instantiate_via_single_import.pkg.U32;

public class InstantiateViaSingleImportTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateViaSingleImport instantiateViaSingleImport = new InstantiateViaSingleImport();
        instantiateViaSingleImport.setTest32(new U32(13));
        instantiateViaSingleImport.setTestStr(new Test_string("test"));

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        instantiateViaSingleImport.write(writer);

        final BitStreamReader reader =
                new ByteArrayBitStreamReader(writer.toByteArray(), writer.getBitPosition());
        final InstantiateViaSingleImport readInstantiateViaSingleImport =
                new InstantiateViaSingleImport(reader);
        assertTrue(instantiateViaSingleImport.equals(readInstantiateViaSingleImport));
    }
}

File: test/language/templates/cpp98/InstantiateViaSingleImportTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_via_single_import/InstantiateViaSingleImport.h"

namespace templates
{
namespace instantiate_via_single_import
{

TEST(InstantiateViaSingleImportTest, readWrite)
{
    InstantiateViaSingleImport instantiateViaSingleImport;
    pkg::U32 u32;
    u32.setValue(13);
    instantiateViaSingleImport.setTest32(u32);
    pkg::Test_string str;
    str.setValue("test");
    instantiateViaSingleImport.setTestStr(str);

    zserio::BitStreamWriter writer;
    instantiateViaSingleImport.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateViaSingleImport readInstantiateViaSingleImport(reader);

    ASSERT_TRUE(instantiateViaSingleImport == readInstantiateViaSingleImport);
}

} // namespace instantiate_via_single_import
} // namespace templates
